src/libultra/libc/ll.c

```diff
@@ -1,47 +1,48 @@
-#include "global.h"
+// IDO Compiler Intrinsics for 64-bit arithmetic
 
-s64 __ull_rshift(u64 l, s64 r) {
+unsigned long long __ull_rshift(unsigned long long l, unsigned long long r) {
     return l >> r;
 }
 
-u64 __ull_rem(u64 l, u64 r) {
+unsigned long long __ull_rem(unsigned long long l, unsigned long long r) {
     return l % r;
 }
 
-u64 __ull_div(u64 l, u64 r) {
+unsigned long long __ull_div(unsigned long long l, unsigned long long r) {
     return l / r;
 }
 
-s64 __ll_lshift(s64 l, s64 r) {
+unsigned long long __ll_lshift(unsigned long long l, unsigned long long r) {
```

Comment:
you changed the types from signed to unsigned here and in other functions below, was that intended?

Reply:
I wanted to better match ultralib, but looking at the libultra debug info the current types in ultralib seem a bit wrong